src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -221,11 +316,30 @@ class SyscallHandler {
   /** The user-space processes running above the kernel. */
   const std::unordered_map<uint64_t, std::shared_ptr<Process>>& processes_;
 
+  /** A memory interface to the system memory. */
+  std::shared_ptr<MemoryInterface> memory_;
+
+  /** A callback function to send a syscall result back to a core through the
+   * SimOS class. */
+  std::function<void(simeng::OS::SyscallResult)> returnSyscall_;
```

Comment:
I think it is worth declaring the callback function type as a typedef as `returnSyscallResult` or anything else it will communicate the intent clearly

Reply:
Happy with the rename suggested

Reply:
I think you have renamed the the variable I meant to typedef the function type instead. Sorry not being clear before.

Reply:
Redid this change